packages/atomic/src/components/common/result-list/grid-display-results.tsx

```diff
@@ -1,53 +1,49 @@
 import {buildInteractiveResult, SearchEngine} from '@coveo/headless';
 import {FunctionalComponent, h} from '@stencil/core';
-import {AnyBindings} from '../interface/bindings';
 import {LinkWithResultAnalytics} from '../../search/result-link/result-link';
-import {ResultsProps} from './result-list-common';
+import {ResultListDisplayProps} from './result-list-common-interface';
+import {extractFoldedResult} from '../interface/result';
 
-export const GridDisplayResults: FunctionalComponent<
-  ResultsProps<AnyBindings>
-> = (props) => {
-  return props.resultListState.results.map((result, index) => {
-    // TODO: support proper engine
+export const GridDisplayResults: FunctionalComponent<ResultListDisplayProps> = (
+  props
+) =>
+  props.getResultListState().results.map((result, index) => {
+    const unFoldedResult = extractFoldedResult(result);
```

Comment:
Small detail, but since [unfolded](https://www.merriam-webster.com/dictionary/unfolded) is an adjective, I wouldn't capitalize the F here.

```suggestion
    const unfoldedResult = extractFoldedResult(result);
```